xlsx/src/export/worksheets.rs

```diff
@@ -280,6 +288,16 @@ pub(crate) fn get_worksheet_xml(
         }
     }
 
+    let merge_cells_section = if merged_cells_count > 0 {
+        format!(
+            "<mergeCells count=\"{merged_cells_count}\">{merged_cells}</mergeCells>",
+            merged_cells_count = merged_cells_count,
+            merged_cells = merged_cells
```

Comment:
This is great but it doesn't match the style I have been doing everywhere:
```rust
    let merge_cells_section = if merged_cells_count > 0 {
        format!(
            "<mergeCells count=\"{}\">{}</mergeCells>",
            merged_cells_count,  merged_cells_str.join("")  )
    } else {
        "".to_string()
    }
```
You don't  need to define `merged_cells` at all. Also I think we should escape all strings.
See producing xlm by hand is dangerous. All strings that come from a user should be escaped `escape_xml(...)`

Reply:
Understood.
removed merged_cells
string escaping added.

Noted the escape_xml() routine.

Reply:
Are the string escaped properly?

Reply:
`        format!(
            "<mergeCells count=\"{}\">{}</mergeCells>",
            merged_cells_count,
            merged_cells_str.join("")
        )`

This has escaped string right ? '"' is escaped with '\' and other than this, I dont see any strings which needs escaping.

Reply:
No, `merged_cells_str.join("")` is itself a string and it is not properly escaped. The string could be something like (it's called xml injection):

```
</mergeCells><some_tag_that_does_something_nasty/>
```

I caused the problem in the first first when I decided to build the xml by hand. That is almost always a bad idea precisely because of this issue. I am not a friend of dependencies and while we do use an external parser to _load_ the xml we build the xml by hand. This does mean we have to be extra careful when building the xml. All strings must be correctly, you can use:

https://github.com/ironcalc/IronCalc/blob/main/xlsx/src/export/escape.rs#L81

for this mater.

Reply:
Apologies for dragging this further,

I tried using the above function ( escape_xml ) on 

`merged_cells_str.join("")`

escaped string o/p : ` "<mergeCells count=\"2\">&lt;mergeCell &quot;ref&quot;=&quot;K7:L10&quot;/&gt;&lt;mergeCell &quot;ref&quot;=&quot;H18:J20&quot;/&gt;</mergeCells>"` 

But getting below error in testcase, when I try to load the same xlsx file again

``Result::unwrap()` on an `Err` value: Xml("Missing \"ref\" XML attribute")`

Plus when I open the exported xlsx file, merged cells are not coming in excel.

I understood purpose of using escaping of xml ( thanks to you ! ), Will this be necessary here ? Because these strings wont be coming from user right ? These strings are made by application and exporting it. Plus the exported XML should have not have these symbols ( &lt, &qut etc ) right ?

Let me know how I can resolve this.

Thanks,
Varun

Reply:
Oh yeah, that wouldn't work :). The `escape_xml` escapes the content inside xml, here we would want to escape an attribute. Now the attribute you are reading from an existing string that has already been imported from Excel. This string should be a range like: `A1:C3`. I have some thouhts about this:

* we shouldn't store strings in the internal structure, maybe we can store already the parsed range?
* this also has implications when importing. So we should import and parse the merge cells ranges

I think to move forward we can merge this as it is (no pun intended) since it is consistent with the importing mechanism.
In an upcoming PR we should fix this though. Either marking cells as part of a merged cells structure or having the merged cells.

By the way your mechanism will work, you just need to escape each of the strings in the refs:

```rust
    for merge_cell_ref in &worksheet.merge_cells {
        merged_cells_str.push(format!("<mergeCell ref=\"{}\"/>", escape_xml(merge_cell_ref)))
    }
```

But let's not do that this time around, we will eventually build that string ourselves from numbers, so there will be no escaping needed.
Many thanks!

Reply:
Understood, I will resolve the conversation here. In the next PR, Lets add this work item as well .